client/lib/post-normalizer/rule-pick-canonical-image.js

```diff
@@ -9,22 +9,26 @@ import { find } from 'lodash';
 import { thumbIsLikelyImage, isCandidateForCanonicalImage } from './utils';
 
 export default function pickCanonicalImage( post ) {
+	let canonicalImage;
 	if ( thumbIsLikelyImage( post.post_thumbnail ) ) {
 		const { URL: url, width, height } = post.post_thumbnail;
-		post.canonical_image = {
+		canonicalImage = {
 			uri: url,
 			width,
 			height
 		};
 	} else if ( post.content_images && post.content_images.length ) {
-		const canonicalImage = find( post.content_images, isCandidateForCanonicalImage );
-		if ( canonicalImage ) {
-			post.canonical_image = {
-				uri: canonicalImage.src,
-				width: canonicalImage.width,
-				height: canonicalImage.height
+		const candidateImage = find( post.content_images, isCandidateForCanonicalImage );
```

Comment:
Just changes it so that it doesn't modify the post object unless it finds an image. 

That way post will still return true for shallow equality checks